credsweeper/file_handler/data_content_provider.py

```diff
@@ -152,28 +154,127 @@ def represent_as_xml(self) -> bool:
             return bool(self.lines and self.line_numbers)
         return False
 
-    def _check_multiline_cell(self, cell: Tag) -> Optional[str]:
-        """multiline cell will be analysed as text or return single line from cell"""
+    def _check_multiline_cell(self, cell: Tag) -> Optional[Tuple[int, str]]:
+        """multiline cell will be analysed as text or return single line from cell
+        returns line number and one line for analysis
+        If there are no text or the text will be analysed as multiline - it returns None"""
         # use not stripped get_text, otherwise all format is cleaned
         cell_text = cell.get_text()
         cell_lines = cell_text.splitlines()
         line_numbers: List[int] = []
         stripped_lines: List[str] = []
-        for line in cell_lines:
+        for offset, line in enumerate(cell_lines):
             if stripped_line := line.strip():
-                line_numbers.append(cell.sourceline)
+                line_numbers.append(cell.sourceline + offset)
                 stripped_lines.append(stripped_line)
         if 0 == len(stripped_lines):
             return None
         elif 1 == len(stripped_lines):
-            return stripped_lines[0]
+            return line_numbers[0], stripped_lines[0]
```

Comment:
It seems returning value `line_numbers[0]` not used anywhere...
Could you please check this again?

Reply:
Yes, it is used 4 times. I've refactored the code to reduce complexity.

Reply:
Ah.. `td_numbered_line[]`.. I see :)